src/raft/replication.c

```diff
@@ -1865,6 +1874,9 @@ int replicationApply(struct raft *r)
 		}
 
```

Comment:
For clarity I would add `entry_sm = NULL; entry = NULL` and move the comment here instead, rephrasing it like: "Note that applyCommand may changed the log and thus the references to `entry` and `entry_sm` are no longer valid."

Reply:
`entry` itself comes from a different structure, managed in a different way and it's not needed to update it.
I moved the comment below. I agree that it could be shortened as what happens with reference counting here is obvious and doesn't need a lecture.